Tests/ApolloCacheDependentTests/FetchQueryTests.swift

```diff
@@ -73,6 +73,65 @@ class FetchQueryTests: XCTestCase, CacheDependentTesting {
     
     wait(for: [serverRequestExpectation, fetchResultFromServerExpectation], timeout: defaultWaitTimeout)
   }
+
+  func testFetchReturningCacheDataOnErrorReturnsDataOnFetchError() throws {
```

Comment:
I would also recommend adding a test here for what error you get when you get an error from the server and there's nothing in the cache so we can keep that behavior consistent

Reply:
Isn't this covered by the `testFetchReturningCacheDataOnErrorReturnsBothErrors` test case?

```swift
      // starting at line 115
      try XCTAssertFailureResult(result) {
        let cacheError = try XCTUnwrap($0 as? LegacyCacheReadInterceptor.CacheError)

        switch cacheError {
        case let .fetchAndCacheFailed(fetchError, cacheError):
          let fetchError = try XCTUnwrap(fetchError as? TestError)
          let cacheError = try XCTUnwrap(cacheError as? JSONDecodingError)
          guard case .missingValue = cacheError else {
            XCTFail("Unexpected json decoding error: \(cacheError)")
            return
          }
          XCTAssertEqual(fetchError, .expectation)
        }
      }
```